hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/DeleteOperation.java

```diff
@@ -207,7 +211,7 @@ public DeleteOperation(final StoreContext context,
         "page size out of range: %d", pageSize);
     this.pageSize = pageSize;
     metadataStore = context.getMetadataStore();
-    executor = context.createThrottledExecutor(2);
+    executor = context.createThrottledExecutor(1);
```

Comment:
what is the intention here? One batched delete at a time per-client?

Reply:
yeah, for now. It means that the delete and list can go in parallel, without having to deal with the complexity of multiple parallel deletes and failures within them. It's the error handling which scared me. And with the same pool of connections to Dynamo, you wouldn't automatically get a speed up. As the Javadoc says: do more experimentation here -but do it on EC2, so that the answers are valid.